src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -73,65 +73,64 @@ private static void CreateRequirements()
             }
         }
 
-        public static async Task<string> ValidatePythonVersion(bool setWorkerExecutable = false, bool errorIfNoExactMatch = false, bool errorOutIfOld = true)
+        public static async Task<WorkerLanguageVersionInfo> ValidatePythonVersion(
+            bool setWorkerExecutable = false,
+            bool errorIfNoExactMatch = false,
+            bool errorOutIfOld = true)
         {
             // If users are overriding this value, we don't have to worry about verification
-            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar)))
+            string pythonDefaultExecutablePath = Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar);
+            if (!string.IsNullOrEmpty(pythonDefaultExecutablePath))
             {
-                return Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar);
+                return await GetVersion(pythonDefaultExecutablePath);
             }
 
-            const string pythonStr = "python";
-            const string python3Str = "python3";
-            const string python36Str = "python3.6";
-            const string py36Str = "3.6";
-            const string py3Str = "3.";
-            const string warningMessage = "Python 3.6.x is recommended, and used in Azure Functions. You are using Python version {0}.";
-            const string errorIfNotExactMessage = "Python 3.6.x is required, and used in Azure Functions. You are using Python version {0}. "
-                + "Please install Python 3.6, and use a virtual environment to switch to Python 3.6.";
-            const string errorMessageOldPy = "Python 3.x (recommended version 3.6.x) is required. Found python versions ({0}).";
-            const string errorMessageNoPy = "Python 3.x (recommended version 3.6.x) is required. No Python versions were found.";
-
-            var pythonExeVersionTask = VerifyVersion(pythonStr);
-            var python3ExeVersionTask = VerifyVersion(python3Str);
-            var python36ExeVersionTask = VerifyVersion(python36Str);
-
-            var pythonExeVersion = await Utilities.SafeExecution(async () => await pythonExeVersionTask) ?? string.Empty;
-            var python3ExeVersion = await Utilities.SafeExecution(async () => await python3ExeVersionTask) ?? string.Empty;
-            var python36ExeVersion = await Utilities.SafeExecution(async () => await python36ExeVersionTask) ?? string.Empty;
-
-            var exeToVersion = new Dictionary<string, string>()
+            const string infoVersionSelectMessage = "We select Python interpreter '{0}' with version {1} for your project.";
```

Comment:
```suggestion
            const string infoVersionSelectMessage = "Found Python {1} ({0})";
```
You might also want to change the ordering (1, 0) in the above suggestion.